src/spec-configuration/containerFeaturesOCI.ts

```diff
@@ -0,0 +1,220 @@
+    id: string;
+    version: string;
+    featureName: string;
+    owner: string;
+    namespace: string;
+    registry: string;
+}
+
+export interface OCIManifest {
+    schemaVersion: number;
+    mediaType: string;
+    config: {
+        mediaType: string;
+        digest: string;
+        size: number;
+    };
+    layers: [
+        {
+            mediaType: string;
+            digest: string;
+            size: number;
+            annotations: {
+                'org.opencontainers.image.ref.name': string;
+                'org.opencontainers.image.title': string;
+            };
+        }
+    ];
+}
+
+export function getOCIFeatureSet(output: Log, identifier: string, options: boolean | string | Record<string, boolean | string | undefined>, manifest: OCIManifest): FeatureSet {
+
+    const featureRef = getFeatureRef(output, identifier);
+
+    const feat = {
+        id: featureRef.id,
+        name: featureRef.id,
+        included: true,
+        value: options
+    };
+
+    let featureSet: FeatureSet = {
+        sourceInformation: {
+            type: 'oci',
+            manifest: manifest,
+            featureRef: featureRef,
+
+        },
+        features: [feat],
+    };
+
+    return featureSet;
+}
+
+export function getFeatureRef(output: Log, identifier: string): OCIFeatureRef {
+
+    // ex: ghcr.io/codspace/features/ruby:1
+    const splitOnColon = identifier.split(':');
+    const id = splitOnColon[0];
+    const version = splitOnColon[1] ? splitOnColon[1] : 'latest';
+
+    const splitOnSlash = id.split('/');
+    const featureName = splitOnSlash[splitOnSlash.length - 1];
+    const owner = splitOnSlash[1];
+    const registry = splitOnSlash[0];
+    const namespace = splitOnSlash.slice(1, -1).join('/');
+
+    output.write(`identifier: ${identifier}`, LogLevel.Trace);
+    output.write(`id: ${id}`, LogLevel.Trace);
+    output.write(`version: ${version}`, LogLevel.Trace);
+    output.write(`featureName: ${featureName}`, LogLevel.Trace);
+    output.write(`owner: ${owner}`, LogLevel.Trace);
+    output.write(`namespace: ${namespace}`, LogLevel.Trace);
+    output.write(`registry: ${registry}`, LogLevel.Trace);
+
+    return {
+        id,
+        version,
+        featureName,
+        owner,
+        namespace,
+        registry
+    };
+}
+
+// Validate if a manifest exists and is reachable about the declared feature.
+export async function validateOCIFeature(output: Log, env: NodeJS.ProcessEnv, identifier: string): Promise<OCIManifest | undefined> {
+    const featureRef = getFeatureRef(output, identifier);
+
+    const manifestUrl = `https://${featureRef.registry}/v2/${featureRef.namespace}/${featureRef.featureName}/manifests/${featureRef.version}`;
+    output.write(`manifest url: ${manifestUrl}`, LogLevel.Trace);
+    const manifest = await getFeatureManifest(output, env, manifestUrl, featureRef);
+
+    if (manifest?.config.mediaType !== 'application/vnd.devcontainers') {
+        return undefined;
```

Comment:
I suggest to add a warning log statement here. Otherwise, if someone publishes a feature and forgets to use this media type they will struggle to figure out what's wrong.